graphql-server/pkg/handlers/auth.go

```diff
@@ -0,0 +1,26 @@
+package handlers
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+
+	"github.com/gin-gonic/gin"
+	"github.com/oursky/likedao/pkg/config"
+	hmacsha256util "github.com/oursky/likedao/pkg/utils/hmacsha256"
+)
+
+//nolint:errcheck
+func NonceHandler(sessionConfig config.SessionConfig) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		nonceBytes := make([]byte, 4)
+		if _, err := rand.Read(nonceBytes); err != nil {
+			c.Error(err)
+			return
+		}
+		nonce := hex.EncodeToString(nonceBytes)
+		signedNonce := hmacsha256util.Sign(sessionConfig.SignatureSecret, nonce)
+
+		c.SetCookie("nonce", signedNonce, sessionConfig.NonceExpiry, "/auth", "", true, true)
```

Comment:
- should this be `nonce.sig`? i think we can also set plaintext nonce at cookies too
- we should set domain as well? to the network hostname `mainnet.likedao.mydomain.com`; otherwise network switch might not work as it defaults to host? [ref mdn docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#domain_attribute)
- better add module prefix? so `auth_nonce`?

Reply:
since we are comparing the nonce in the message directly from the nonce response we probably don’t need it in the cookie but i guess it can be a reference

Reply:
yes, for debug; & previously i thought we can add a middleware / abstraction to read / set cookies with signature by default; so filter away unsigned pairs at read & set `.sig` at write